SpatialGDK/Source/Private/Interop/Connection/SpatialWorkerConnection.cpp

```diff
@@ -124,18 +130,24 @@ void USpatialWorkerConnection::ConnectToLocator()
 	Worker_DeploymentListFuture_Get(DeploymentListFuture, nullptr, this,
 		[](void* UserData, const Worker_DeploymentList* DeploymentList)
 	{
+		USpatialWorkerConnection* SpatialConnection = static_cast<USpatialWorkerConnection*>(UserData);
+
 		if (DeploymentList->error != nullptr)
 		{
-			UE_LOG(LogSpatialWorkerConnection, Error, TEXT("Error fetching deployment list: %s"), UTF8_TO_TCHAR(DeploymentList->error));
+			FString ErrorMessage = FString::Printf(TEXT("error fetching deployment list: %s"), UTF8_TO_TCHAR(DeploymentList->error));
+			UE_LOG(LogSpatialWorkerConnection, Error, TEXT("Failed to connect to SpatialOS: %s"), *ErrorMessage);
+			SpatialConnection->OnConnectFailed.ExecuteIfBound(ErrorMessage);
+			return;
 		}
 
 		if (DeploymentList->deployment_count == 0)
 		{
-			UE_LOG(LogSpatialWorkerConnection, Error, TEXT("Received empty list of deployments. Failed to connect."));
+			FString ErrorMessage = FString::Printf(TEXT("received empty list of deployments."));
```

Comment:
Nitnitnit: `Received`

Reply:
Made it lowercase on purpose so it can be concatenated to an error message prefix in the OnConnectFailed handler. Happy to change back if you feel strongly about it.